kolibri/core/discovery/tasks.py

```diff
@@ -372,11 +404,21 @@ def reset_connection_states(broadcast_id):
         connection_faults=0,
     )
 
-    # enqueue update tasks for all static locations
-    for static_location_id in StaticNetworkLocation.objects.all().values_list(
-        "id", flat=True
+    # enqueue update tasks for all static locations except KDP
+    for static_location_id in (
+        StaticNetworkLocation.objects.all()
+        .values_list("id", flat=True)
+        .exclude(id=DATA_PORTAL_BASE_INSTANCE_ID)
```

Comment:
What was the reasoning behind this?

Reply:
The line `.exclude(id=DATA_PORTAL_BASE_INSTANCE_ID)` was added to prevent enqueuing a `perform_network_location_update` task for KDP network locations.

Reply:
Gotcha, but I believe KDP shouldn't be returned here, because this uses the proxy model `StaticNetworkLocation`, which should filter on `location_type = static`. Theoretically, a user could add KDP as a static location, which this would catch, but as long as things still work, I wouldn't worry about that.